### DISCUSSION TITLE
Immediately update multiple figures in an alternating fashion

### DISCUSSION THREAD

**Discussion by denniskb** [alandefreitas/matplotplusplus#244] [Q&A] [answered]
Thanks for this great project!

I'm trying to render an "animation" using `figure()` and `scatter()` (basically updating the plot 10x per sec with new data). This works flawlessly with a single figure. However, as soon as I use 2+ figures, they no longer update immediately and I can't seem to force them to. `h->draw()` has seemingly no effect (I only see the figures after the last `scatter()` call). Both `show()` as well as `show(h)` work in that they update the figures immediately, but they halt the app until I press ENTER. I'd really love to have the figures update without user interaction so I can monitor a process over time.

Here's my basic setup:
```
auto h1 = figure();
auto h2 = figure();
for (...) {
  // compute data
  figure(h1);
  scatter(...);
  figure(h2);
  scatter(...);
}
```

**Comment by denniskb** [comment, marked as answer]
Using quiet mode `figure(true)` in combination with manually drawing the figure `h->draw()` seems to do the trick :)

**Comment by alandefreitas** [comment]
Exactly. `show()` is meant to block the calling thread so you can see the figure. `draw()`, or any other function in non-quiet mode, is meant to update the figure.
